charts/prometheus/templates/pdb.yaml

```diff
@@ -11,5 +11,10 @@ spec:
   selector:
     matchLabels:
       {{- include "prometheus.server.matchLabels" . | nindent 6 }}
-  {{- toYaml $pdbSpec | nindent 2 }}
+  {{- if $pdbSpec.minAvailable }}
+  minAvailable: {{ $pdbSpec.minAvailable }}
+  {{- end }}
+  {{- if $pdbSpec.maxUnavailable }}
+  maxUnavailable: {{ $pdbSpec.maxUnavailable }}
+  {{- end }}
```

Comment:
We can make PDB a bit friendlier to users who would like to set `minAvailable` since they have to set two fields: unset default `maxUnavailable` and then set `minAvailable`. I would suggest a change in this respect - commenting the current default out in the values file and rewriting the template while setting the current default. `unhealthyPodEvictionPolicy`should not be discarded.

```yaml
  {{- if not (or (hasKey $pdbSpec "minAvailable") (hasKey $pdbSpec "maxUnavailable")) }}
  maxUnavailable: 1
  {{- end }}
  {{- if hasKey $pdbSpec "minAvailable" }}
  minAvailable: {{ $pdbSpec.minAvailable }}
  {{- end }}
  {{- if hasKey $pdbSpec "maxUnavailable" }}
  maxUnavailable: {{ $pdbSpec.maxUnavailable }}
  {{- end }}
  {{- if hasKey $pdbSpec "unhealthyPodEvictionPolicy" }}
  unhealthyPodEvictionPolicy: {{ $pdbSpec.unhealthyPodEvictionPolicy }}
  {{- end }}
```

We'd have to add a comment in the values stating that default is set to `maxUnavailable=1` if neither `minAvailable` nor `maxUnavailable` is set. What dou you think?

Reply:
Thank you, @mentlak0. Please, comment out the default value in the values file L456.

Reply:
@zeritti Done